plugins/auth-backend/src/lib/flow/authFlowHelpers.ts

```diff
@@ -31,8 +31,15 @@ export const postMessageResponse = (
 
   // TODO: Make target app origin configurable globally
   const script = `
-    (window.opener || window.parent).postMessage(JSON.parse(atob('${base64Data}')), '${appOrigin}')
-    window.close()
+    var json = atob('${base64Data}');
+    var origin = '${appOrigin}';
+    try {
+      var encoded = Uint8Array.from(json, c => c.codePointAt(0));
```

Comment:
Another option would be to pass a base64 encoded data: uri to a Blob, but that code is much longer and has the same IE problems... If the current solution works I would stay with it.

But should we add a comment that says why we have this trick in place?

Reply:
Sure, maybe above then, on like 32. Don't have to send it over the wire :) I'll fix

Reply:
@Fox32 @Rugvip Uh this made me wonder - why don't we just use encode/decodeURIComponent instead that are perfectly aware of non-7-bit chars and are extremely widely supported? Updated the PR accordingly

Reply:
@freben Because injecting user controlled content into html on a secure domain is super scary and I want to be really far away from any possibilities of injection rather than treading the line. The current implementation with `encodeURIComponent` is broken and imo it's too easy to revert to the mistake even if we fix it. I'd prefer sticking to base64 and decoding though an `UInt8Array` and `TextDecoder`.

Reply:
In what way is it broken? I didn't get that part.

Reply:
In terms of

- how could `encodeURIComponent` result in any character that isn't safe to put in the document as we do
- how could `decodeURIComponent` result in a string that somehow breaks `JSON.parse` in a dangerous way
- how could UTF characters be misrepresented in `decodeURIComponent(encodeURIComponent(x))`

Reply:
Ah haha, I forgot about it not encoding `'` but encoding `"`. So I would still like to use this method but doing `var data = "${data}"`

Reply:
Yup, but let's not. With that you gonna slap a big comment above it explaining why it is the way it is and why nobody should change it back to `'`. And even with that there's plenty of occasions of comments not being read :grin:. Better just stick to a safer encoding.

Reply:
This gives you some decent browser support without being very verbose:

```
new TextDecoder().decode(Uint8Array.from(Array.from(atob('8J+Ukg'), c => c.charCodeAt(0))))
```

Reply:
Yeah that's what the code used to do right before you reviewed, it's updated now with a wall of text accompanying it

Reply:
I avoided `Uint8Array.from` because it doesn't have as broad support as `Array.from`

Reply:
Uh derp, except that I didn't, this would be the correct one:

```
new TextDecoder().decode(new Uint8Array(Array.from(atob('8J+Ukg'), c => c.charCodeAt(0))))
```

Reply:
Ah I didn't know about the `from` support - updated to use `Array.from`

Reply:
Still kinda wondering if `encodeURIComponent(x).replace(/'/g, '%27')` wouldn't have been way better and easier and well supported and ... :)

Reply:
... scarier, but do it if you want to. I can't think of anything that could go wrong with it right now